Modules/SubModule_Import/xASL_bids_BIDS2Legacy.m

```diff
@@ -33,48 +32,52 @@
 % 6. Clean up
 % 
 % -----------------------------------------------------------------------------------------------------------------------------------------------------
-% EXAMPLE: [x] = xASL_bids_BIDS2Legacy(pathStudy, x, bOverwrite);
+% EXAMPLE: [x] = xASL_bids_BIDS2Legacy(x, bOverwrite);
 % __________________________________
 % Copyright 2015-2021 ExploreASL
 
 
 %% 0. Admin
 
 % Verify the input parameters
-if nargin<1 || isempty(pathStudy)
-	error('pathStudy is a required parameter.');
+if isempty(x.dir.DatasetRoot)
+	error('x.dir.DatasetRoot is a required parameter.');
 end
 
-if nargin<3 || isempty(bOverwrite)
+if nargin<2 || isempty(bOverwrite)
     bOverwrite = 1;
 end
 
 % Verify that the rawdata subfolder exists
-if ~exist(fullfile(pathStudy,'rawdata'),'dir')
+if ~exist(fullfile(x.dir.DatasetRoot,'rawdata'),'dir')
     warning('Invalid folder selected, not containing rawdata folder');
     return;
 end
 
+% Check derivatives of ExploreASL
+if ~isfield(x.dir,'xASLDerivatives')
```

Comment:
Note that nobody knows what "xASL" means, that is just an abbreviation that we use internally. Best to replace this by "ExploreASL derivatives folder" or something that doesn't lead to the user asking you what you meant ;)

Reply:
This should only occur if it really crashes. In that case it doesn't make sense for the user to know that one x field which isn't called "ExploreASL derivatives folder" is missing. 

We should differentiate between warnings for users and warnings for developers.